### PR TITLE
update BitlyProvider to v4 API

### PR DESCRIPTION
**Pull request by navitern**

### CODE REVIEW

**Review by mcaskill — COMMENTED** [review]
Unfortunately, these changes do not work. The V4 API requires POST requests with payload in the body.

File: src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php [hunk 1]
```diff
@@ -65,8 +65,9 @@ public function shorten(LinkInterface $link, $domain = null)
     {
         $client = $this->createClient();
 
-        $response = $client->get(sprintf('/v3/shorten?access_token=%s&longUrl=%s&domain=%s',
-            $this->auth->getAccessToken(),
+        $this->options['headers']['Authorization'] = 'Bearer ' . $this->auth->getAccessToken();
+
+        $response = $client->get(sprintf('/v4/shorten?longUrl=%s&domain=%s',
```
Comment (mcaskill) [inline src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php:70]:
```php
$response = $client->post('/v4/shorten', array_merge(
    array(
        'json' => array(
            'domain'   => $domain,
            'long_url' => $link->getLongUrl(),
        ),
    ),
    $this->options
));

$response = $this->validate($response->getBody()->getContents(), $response->getStatusCode());
```

File: src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php [hunk 2]
```diff
@@ -88,8 +89,9 @@ public function expand(LinkInterface $link, $hash = null)
     {
         $client = $this->createClient();
 
-        $response = $client->get(sprintf('/v3/expand?access_token=%s&shortUrl=%s&hash=%s',
-            $this->auth->getAccessToken(),
+        $this->options['headers']['Authorization'] = 'Bearer ' . $this->auth->getAccessToken();
+
+        $response = $client->get(sprintf('/v4/expand?shortUrl=%s&hash=%s',
```
Comment (mcaskill) [inline src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php:94]:
```php
$response = $client->post('/v4/expand', array_merge(
    array(
        'json' => array(
            'bitlink_id' => $hash ?: $link->getShortUrl(),
        ),
    ),
    $this->options
));

$response = $this->validate($response->getBody()->getContents(), $response->getStatusCode());
```

**Review by mcaskill — CHANGES_REQUESTED** [review]
The `validate()` also needs to be updated accordingly:

```php
/**
 * Validates the Bit.ly's response and returns it whether the status code is 200.
 *
 * @param string  $apiRawResponse
 * @param integer $statusCode
 *
 * @return object
 *
 * @throws InvalidApiResponseException
 */
private function validate($apiRawResponse, $statusCode)
{
    $response = json_decode($apiRawResponse);

    if (null === $response) {
        throw new InvalidApiResponseException('Bit.ly response is probably mal-formed because cannot be json-decoded.');
    }

    if ($statusCode < 200 || $statusCode >= 300) {
        throw new InvalidApiResponseException(sprintf('Bit.ly returned status code "%s" with message "%s"',
            $statusCode,
            property_exists($response, 'message') ? $response->message : ''
        ));
    }

    return $response;
}
```

**Comment by navitern** [comment]
Yes, my mistake.

**Pull request closed without merging**